### PR TITLE
UCP/PROTO/STREAM: Add stream/zcopy/multi protocol

### PR DESCRIPTION
**Pull request by yosefe**
## Why
Enable zero-copy for stream API protov2

## How
- Added stream/multi/zcopy protocol definitions
- Added `ucp_proto_eager_zcopy_multi_common_send_func()`, reuse it in stream, tag/eager, and rndv/data protocols

### CODE REVIEW

**Comment by yosefe** [comment]
@rakhmets @brminich can you pls take a look?

**Review by brminich — APPROVED** [review]

File: src/ucp/proto/proto_multi.inl [hunk 1]
```diff
@@ -307,4 +307,38 @@ ucp_proto_eager_bcopy_multi_common_send_func(
     return ucp_proto_bcopy_send_func_status(packed_size);
 }
 
+static UCS_F_ALWAYS_INLINE ucs_status_t
+ucp_proto_eager_zcopy_multi_common_send_func(
```
Comment (rakhmets) [inline src/ucp/proto/proto_multi.inl:311]:
```suggestion
ucp_proto_zcopy_multi_common_send_func(
```

File: src/ucp/stream/stream_multi.c [hunk 2]
```diff
@@ -18,6 +18,13 @@
 /* Convenience macros for setting eager protocols descriptions  */
 #define UCP_PROTO_STREAM_BCOPY_DESC \
     "stream " UCP_PROTO_COPY_IN_DESC " " UCP_PROTO_COPY_OUT_DESC
+#define UCP_PROTO_STREAM_ZCOPY_DESC \
+    "stream " UCP_PROTO_ZCOPY_DESC " " UCP_PROTO_COPY_OUT_DESC
+
+static int ucp_stream_check_op(const ucp_proto_init_params_t *init_params)
+{
+    return ucp_proto_init_check_op(init_params, UCS_BIT(UCP_OP_ID_STREAM_SEND));
+}
```
Comment (rakhmets) [inline src/ucp/stream/stream_multi.c:27]: Sorry, didn't noticed that `ucp_proto_multi_init` is also the same in both initialize methods. 
```suggestion
static ucs_status_t
ucp_stream_multi_init(const ucp_proto_multi_init_params_t *params)
{
    if (!ucp_proto_init_check_op(&params->super.super,
                                 UCS_BIT(UCP_OP_ID_STREAM_SEND))) {
        return UCS_ERR_UNSUPPORTED;
    }

    return ucp_proto_multi_init(params, params->super.super.priv,
                                params->super.super.priv_size);
}
```

, and 
```C
static ucs_status_t
ucp_stream_multi_bcopy_init(const ucp_proto_init_params_t *init_params)
{
    ...
    return ucp_stream_multi_init(&params);
}
...
static ucs_status_t
ucp_stream_multi_zcopy_init(const ucp_proto_init_params_t *init_params)
{
    ...
    return ucp_stream_multi_init(&params);
}
```

File: src/ucp/stream/stream_multi.c [hunk 3]
```diff
@@ -18,6 +18,13 @@
 /* Convenience macros for setting eager protocols descriptions  */
 #define UCP_PROTO_STREAM_BCOPY_DESC \
     "stream " UCP_PROTO_COPY_IN_DESC " " UCP_PROTO_COPY_OUT_DESC
+#define UCP_PROTO_STREAM_ZCOPY_DESC \
+    "stream " UCP_PROTO_ZCOPY_DESC " " UCP_PROTO_COPY_OUT_DESC
```
Comment (rakhmets) [inline src/ucp/stream/stream_multi.c:22]:
```suggestion
#define UCP_STREAM_DESC "stream"
#define UCP_PROTO_STREAM_BCOPY_DESC \
    UCP_STREAM_DESC " " UCP_PROTO_COPY_IN_DESC " " UCP_PROTO_COPY_OUT_DESC
#define UCP_PROTO_STREAM_ZCOPY_DESC \
    UCP_STREAM_DESC " " UCP_PROTO_ZCOPY_DESC " " UCP_PROTO_COPY_OUT_DESC
```

**Pull request merged**